src/wasm/wasm-type.cpp

```diff
@@ -86,7 +86,7 @@ struct HeapTypeInfo {
   // global store.
   bool isTemp = false;
   bool isOpen = false;
-  bool isShared = false;
+  Shareability share = Unshared;
```

Comment:
This internal flag could maybe be a bool just for consistency with the two above it, but if it's more convenient otherwise sgtm as is.

Reply:
I'd rather avoid mixing bool and `Shareability` to avoid future questions of whether we should use one or the other in specific cases.